### PR TITLE
add multi Y options to nipals

### PR DESCRIPTION
**Pull request by josoriom**

### CODE REVIEW

**Comment by lpatiny** [comment]
Could you add an explanation about why exactly you want this change ? I don't see difference between. before and after.

**Comment by josoriom** [comment]
First I removed this part of the code that disables the possibility of an Y with more than one column.
https://github.com/mljs/matrix/blob/master/src/dc/nipals.js#L21-L24

error message for when the maximum number of iterations is reached
https://github.com/josoriom/matrix/blob/master/src/dc/nipals.js#L48-L50

the normalization of loadings as the source code
https://github.com/josoriom/matrix/blob/master/src/dc/nipals.js#L55
source code: https://github.com/jobo322/MetaboMate/blob/master/R/NIPALS_PLS_component.R#L51

and the testcase to compare the results of both codes:
https://github.com/josoriom/matrix/blob/master/src/__tests__/decompositions/nipals.test.js#L97-L124

File: src/dc/nipals.js [hunk 1]
```diff
@@ -18,22 +18,15 @@ export default class nipals {
       } else {
         Y = WrapperMatrix2D.checkMatrix(Y);
       }
-      if (!Y.isColumnVector() || Y.rows !== X.rows) {
```
Comment (jwist) [inline src/dc/nipals.js:21]: I would leave the test, just adapt it so it works for multiple Y

File: src/dc/nipals.js [hunk 2]
```diff
@@ -18,22 +18,15 @@ export default class nipals {
...
-        throw new Error('Y must be a column vector of length X.rows');
```
Comment (jwist) [inline src/dc/nipals.js:22]: adapt the text of the error message as well to reflect the results of the new test

File: src/dc/nipals.js [hunk 3]
```diff
@@ -47,16 +40,19 @@ export default class nipals {
...
-    }
+      counter++;
+      if (counter === maxIterations) {
+        throw new Error('The maximum number of iterations has been reached');
+      }
+    } while (diff > terminationCriteria);
 
     if (Y) {
       let p = X.transpose().mmul(t).div(t.transpose().mmul(t).get(0, 0));
-      p = p.div(p.norm());
```
Comment (jwist) [inline src/dc/nipals.js:59]: I would leave the normalisation as before, I think this is wrong in the reference code

**Review by jwist — CHANGES_REQUESTED** [review]
it looks fine to me. The difference is now that we can use a responde vector of more than one dimension, i.e., with multiple columns

**Pull request merged**